packages/toolpad-app/src/ThemeContext.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { ThemeProvider as MuiThemeProvider, createTheme } from '@mui/material/styles';
+import { deepmerge } from '@mui/utils';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/material/utils';
+import { getDesignTokens, getThemedComponents, getMetaThemeColor } from './theme';
+
+interface ThemeProviderProps {
+  children?: React.ReactNode;
+}
+
+export type PaletteModeOptions = 'light' | 'dark';
+
+export const DispatchContext = React.createContext<
+  React.Dispatch<React.SetStateAction<PaletteModeOptions>>
+>(() => {
+  throw new Error('Forgot to wrap component in `ThemeProvider`');
+});
+
+export function ThemeProvider(props: ThemeProviderProps) {
+  const { children } = props;
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+  const preferredMode = prefersDarkMode ? 'dark' : 'light';
+  const [paletteMode, setPaletteMode] = React.useState<PaletteModeOptions>(preferredMode);
+
+  React.useEffect(() => {
+    if (typeof window !== 'undefined') {
+      setPaletteMode(
+        JSON.parse(localStorage.getItem('paletteMode') as PaletteModeOptions) || preferredMode,
```

Comment:
Can we use our `useLocalStorage` hook?

Reply:
That is what is being used to set the value; if I'm not wrong there's no good way to just get the value of a local storage variable without setting anything to it using the hook - so I went ahead with using this.

Reply:
The hook is not setting anything in localstorage until you actually call its returned `setState` function.
The hook will be a lot smarter than what this effect does. (e.g. sync across tabs, deal better with SSR)

Reply:
You're right - this works well. I was probably making an unknown error in the first implementation I did with `useLocalStorageState` (I imagine the issue was defining it twice, instead of wrapping it in one function and importing it in both places which is smarter)